tendrl/commons/flows/unmanage_cluster/__init__.py

```diff
@@ -14,59 +21,203 @@ def run(self):
+                    'job_name': self.__class__.__name__,
+                    'status': "in_progress"
+                }
+                _cluster.save()
+
             super(UnmanageCluster, self).run()
-            _cluster = NS.tendrl.objects.Cluster(
-                integration_id=integration_id
-            ).load()
-            _cluster.status = ""
-            _cluster.is_managed = "no"
-            _cluster.locked_by = {}
-            _cluster.errors = []
-            _cluster.current_job = {
-                'status': "finished",
-                'job_name': self.__class__.__name__,
-                'job_id': self.job_id
-            }
-            _cluster.save()
+
+            # Below code to be executed on the parent node.
+
+            _job = NS.tendrl.objects.Job(job_id=self.job_id).load()
+            if 'parent' not in _job.payload:
+                # Creating job to delete monitoring details.
+                _job_id = str(uuid.uuid4())
+                payload = {
+                    "tags": ["tendrl/integration/monitoring"],
+                    "run": "monitoring.flows.DeleteMonitoringData",
+                    "status": "new",
+                    "parameters": self.parameters,
+                    "parent": self.parameters['job_id'],
+                    "type": "monitoring"
+                }
+                NS.tendrl.objects.Job(
+                    job_id=_job_id,
+                    status="new",
+                    payload=payload
+                ).save()
+
+                # Wait for 2 mins for the job to complete
+                loop_count = 0
+                wait_count = 24
+                while True:
+                    if loop_count >= wait_count:
+                        logger.log(
+                            "error",
+                            NS.publisher_id,
+                            {
+                                "message": "Clearing monitoring data for "
+                                           "cluster (%s) not yet complete. "
+                                           "Timing out." %
+                                           NS.tendrl.objects.Cluster(
+                                               integration_id=integration_id
+                                           ).load().short_name
+                            },
+                            job_id=self.parameters['job_id'],
+                            flow_id=self.parameters['flow_id'],
+                        )
+                        return False
+                    time.sleep(5)
+                    finished = True
+                    job = NS.tendrl.objects.Job(job_id=_job_id).load()
+                    if job.status != "finished":
+                        finished = False
+                    if finished:
+                        break
+                    else:
+                        loop_count += 1
+                        continue
+
+                # Deleting cluster details
```

Comment:
Create a separate function for this and invoke here

Reply:
Will do it